Scenes/Towers/Autocast.gd

```diff
@@ -434,6 +435,22 @@ func _get_target_for_buff_autocast() -> Unit:
 	return null
 
 
+func _filter_target_units_for_caster_buff_group(caster: Unit, targets: Array) -> Array:
+	var caster_outgoing_buff_groups: Array[String] = caster.get_buff_groups(BuffGroup.Mode.OUTGOING)
+	
+	if caster_outgoing_buff_groups.is_empty():
+		return targets
+	
+	var filtered_targets: Array = []
+	for target in targets:
+		for target_group in target.get_buff_groups(BuffGroup.Mode.INCOMING):
+			var group_number: int = BuffGroup.get_buff_group_number(target_group)
+			if caster_outgoing_buff_groups.has(group_number):
```

Comment:
This line causes a runtime error because
caster_outgoing_buff_groups = array of strings like "1_OUTGOING", "5_OUTGOING"...
group_number = integer

I think you need to convert caster_outgoing_buff_groups to array of ints of outgoing groups.